### DISCUSSION TITLE
Adding new coordinate systems

### DISCUSSION THREAD

**Discussion by varchasgopalaswamy** [AMReX-Codes/amrex#2755] [Q&A]
Hi,

I'm interested in working with coordinate systems that don't seem to have support right now in AMREX. A simple example of the sort of thing I'd be interested in would be 3D spherical or toroidal coordinates

It looks like the coordinate information is provided entirely by AMReX_CoordSys.cpp. Is there a way I can subclass `CoordSys`, modify it to provide the volumes/surface areas for the coordinate system I'm interested in, and provide the modified `CoordSys` class to the code?

**Comment by etpalmer63** [comment]
Hi,

For spherical coordinates, I can save you some work :smile: . Those have already been implemented in AMReX and are used by some of our application codes. You can find more information about it in the User's Guide [here](https://amrex-codes.github.io/amrex/docs_html/Basics.html?highlight=spherical#realbox-and-geometry). 

For the other parts of your question I will need to defer to someone who knows more about the geometry development.

Thanks for your interest, 

Erik

**Reply by colinmcnally** [reply]
Hi,

Is there a pointer you can provide to a application codes use of AMReX spherical coordinates in more than 1D? 
I found 1D spherical usage in Castro, but nothing beyond that in my searches.

Thanks,
-Colin

**Comment by WeiqunZhang** [comment]
If it's standard spherical coordinates, please feel free to modify CoordSys and submit a PR.  That being said, there are very few places in amrex that use the volume/area functions in CoordSys.  Average-down weighted by volume is an example.  However, it should be straightforward to have your own version of that without using CoordSys.
